Fixes, and new features

tenzap opened the pull request:
It would be nice if this could be merged before other PRs appear since there are many changes (in order to prevent conflicts).

The state of the code in the PR was used to submit to OSM the "Fluo Grand Est 68" bus network in France.

- Permit to open & build again on Netbeans
- GUI fixes & new processing options (see screenshot)
- Support zip files on windows (with content type application/x-zip-compressed
- Support also regex for operator alias
- Stops:
  - Permit to add a new Stop if there is no satisfactory match (closes #24)
  - Support also platforms stored as `ways` in OSM (until now, only nodes were supported)
  - Support using stop names & altnames from a Netex export if it is given as input. When this is used, the name of the stop will be prefixed by the name of the town, and the simple name is put to alt_names. It is possible to skip this for the citied given in the UI which is useful in large cities where we don't want the stops to be prefixed by the town.
- Routes:
  - set role of GTFS stops/platforms to platform instead of stop (which was incorrect)
  - replace gtfs_route_id by gtfs:route_id (gtfs:route_id is much more common in OSM)
  -  Add support for PTv2 on routes (possible to switch between ptv1 / ptv2 by a processing option)
  - Create a single route for each GTFS trip (a trip is known as a route variant or route in OSM)
- Various small fixes.

![Capture d'écran 2023-07-31 200932](https://github.com/CUTR-at-USF/gtfs-osm-sync/assets/46226844/a4879ddd-35a7-41b6-a58f-ef0b149ac2b7)

tenzap committed:
replace gtfs_route_id by gtfs:route_id (through tag_defs.OSM_ROUTE_ID_KEY)

tenzap committed:
use names & altnames from Netex

tenzap committed:
fix modeType on stops based on public_transport value

tenzap committed:
query also 'ways' being platforms

tenzap committed:
fix crash when arr is null

tenzap committed:
when adding platform to route, set member type to 'platform' instead of 'stop'

tenzap committed:
CompareData: Stops: when at the same position, only consider equal when both gtfs & osm have the same type

Same type means either both are platforms, or both are stations

Otherwise, we may match a GTFS station with a OSM platform/stop, or a GTFS platform/stop with an OSM station
which would be wrong.

This was especially a problem when the GTFS station & GTFS stop/platform are
at the exact gps position.

tenzap committed:
add support for 'way' primitives that hold plaforms

tenzap committed:
move to 'netbeans maven' instead of ant project

tenzap committed:
support zip file of content type application/x-zip-compressed

tenzap committed:
Routes: add support for route variants - part1

tenzap committed:
write gtfs route name to tag_defs.OSM_ROUTE_NAME_KEY (ie. gtfs:name) and not name

in OSM, 'name' tag has a specific format which is not equivalent to the
route name in GTFS

tenzap committed:
convert HashSet to LinkedHashSet for relation members

Because order is important in ptv2

tenzap committed:
fix form block guard error

tenzap committed:
add local maven repo with swingx-beaninfo 1.6 and use it + add other maven deps

tenzap committed:
remove now useless swingx-ws-1.0-bin dir

tenzap committed:
remove line that causes problems when editing form in netbeans

Error was:
Error in loading component property: [JFrame]->jTabbedPane1->jPanel1->mapJXMapKit->scrollableTracksViewportWidth
No such property exists in the component.

tenzap committed:
fix ReportViewer form to work again on Netbeans

Some older commits broke the possibility to update the gui in netbeans
and the autogeneration of that code.

This will restore it.

Tested on netbeans 18

tenzap committed:
Routes: routevariant: add gtfsTripIdToRouteVariantMap & allRouteVariantsByRoute

tenzap committed:
update MainForm to be able to edit it in Netbeans again

tenzap committed:
OperatorInfo: add regex support (part1: hardcoded regex)

tenzap committed:
Support Netex selection from UI

tenzap committed:
OperatorInfo: permit to set regex in GUI (part2: when set from GUI)

tenzap committed:
rename some tag_defs and add others

tenzap committed:
Small simplification when parsing gtfs routes

Possible now that we have updated tag_defs.java

tenzap committed:
Routes: don't add gtfs_route_text_color gtfs_agency_id to the routes

By not saving them when reading the routes.txt gtfs file

tenzap committed:
add action='modify' to .dsc file

This is required so that when opening the DSC file in JOSM it is possible to
send the changeset (especially when there are no modifications but only
creations in the dsc)

tenzap committed:
trim & remove duplicate spaces in values read from Gtfs when storing as Osm value

If values are not trimmed or have duplicate spaces, jsom will warn on upload
of the osc file

tenzap committed:
Routes: compareRouteData: migrate to tripId as key

tenzap committed:
Routes: migrate to tripId as key in ReportViewer

tenzap committed:
add ProcessingOptions enum (empty for now)

tenzap committed:
MainForm: move options to stopOptionsPanel & routeOptionsPanel

tenzap committed:
MainFrom for Routes: add checkboxes for options

tenzap committed:
Routes: take into account the selected new route options from UI

tenzap committed:
MainForm: fix width of option panels

tenzap committed:
MainForm: split route options in route & member options

And enable by default the options required for PTv2

tenzap committed:
fix merge of multiple osm values for a key (separated by ';')

This will avoid duplicate values, and also trim the individual values

tenzap committed:
use default selection colors in dataTables

tenzap committed:
Stops: skip GTFS stations - part1

tenzap committed:
MainForm: left align StopOptions components

tenzap committed:
Stops: skip GTFS stations - part2 (ui)

tenzap committed:
Stops & Netex: don't prefix station name with TOWN for some cities - part 1

tenzap committed:
Stops & Netex: don't prefix station name with TOWN for some cities - part 2 (UI)

tenzap committed:
Routes: save stop name with town prefix

Useful to build the route name with the name of the town for 'from' & 'to'

tenzap committed:
MainForm: allow to filter boundingbox for OSM query

Sometimes there are stops which are on an abnormal location which would
increase the boudingbox for the osm query, which could become too large,
For example if the stop if on another continent...

tenzap committed:
ReportViewer: replace Stop by Route in label for count of Routes

tenzap committed:
ReportView: Bus Tab: update radiobutton labels & nextbutton label

tenzap committed:
ReportViewer: give focus to GtfsRoutesCombobox when accepting a route

tenzap committed:
CompareData Routes: improve detection of gtfsId of members

tenzap committed:
RouteMemberTableCellRenderer: set different bg color if identical values

tenzap committed:
Routes: Don't add gtfs stops as members if they are not in OSM (don't have osm id)

tenzap committed:
Stops: permit to add a new stop if the potential osm matches are not satisfactory

tenzap committed:
Routes: permit to choose between PTv2 or PTv1

tenzap committed:
in dataTable, when both gtfs/osm values are empty, set new value as empty too

tenzap committed:
Routes: group stop_positions & platforms in the routes/relations for PTv2

tenzap committed:
ReportViewer: Route: add warning that stops must be added before updating routes

tenzap committed:
consider FFFFFF route color as not set

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/CUTR-at-USF/gtfs-osm-sync?pullRequest=62) <br/>All committers have signed the CLA.

tenzap committed:
netex: Support format provided by Mobigo (FR), throw exception if unsupported format

tenzap commented:
What do you mean by full way/relation support? Is it in addition to what's in this MR?

tenzap committed:
Routes: fix gtfs:route_id which was incorrectly set to trip_id:sample

tenzap committed:
Routes: add OSM tags also to existing routes

Otherwise, in the reportview the osm tags are not copied to the 'new value' column

tenzap committed:
Routes: detect if members differ, in which case display it in ReportView

tenzap committed:
Stops: fix regression where last edited info wan't displayed anymore

Regression appeared in e8e89ed6fe6c28b680b4c1b272f2b28c3815c802

tenzap committed:
Stops: ReportViewer: display location of OSM & Gtfs stop for 'MODIFY' category

tenzap committed:
Stops: Netex: Improve alt_name & name, fix œ detection

In some cases, town with "œ" were not considered same as when it has "oe"
There were also problems where alt_names had too much values (duplicates because
of accent or œ character)
